File: httpserver/src/main/java/com/github/mjdev/libaums/server/http/server/HttpServer.java
package com.github.mjdev.libaums.server.http.server;

import com.github.mjdev.libaums.server.http.UsbFileHttpServer;
import com.github.mjdev.libaums.server.http.UsbFileProvider;

import java.io.IOException;

/**
 * Created by magnusja on 16/12/16.
 */
public interface HttpServer {
    void start() throws IOException;
    void stop() throws IOException;
    boolean isAlive();
    String getHostname();
    int getListeningPort();
    void setUsbFileProvider(UsbFileProvider provider);
}

File: httpserver/src/main/java/com/github/mjdev/libaums/server/http/server/HttpServer.kt
package com.github.mjdev.libaums.server.http.server

import com.github.mjdev.libaums.server.http.UsbFileProvider
import java.io.IOException

/**
 * Created by magnusja on 16/12/16.
 */
interface HttpServer {
    val isAlive: Boolean
    val hostname: String
    val listeningPort: Int
    var usbFileProvider: UsbFileProvider
    @Throws(IOException::class)
    fun start()

    @Throws(IOException::class)
    fun stop()
}

